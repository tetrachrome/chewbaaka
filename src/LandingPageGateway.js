/**
 * LandingPageGateway.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 02, 2020
 */

import React from 'react';

import "semantic-ui-css/semantic.min.css";

import {  Button, Container, Header, Segment } from 'semantic-ui-react'

import './LandingPageGateway.css';

class LandingPageGateway extends React.Component {
  render() {
    return (
      <Segment vertical textAlign="center" className={this.centerSegmentStyleClassName()}>
        <Container className="content">
          <Header inverted as="h1">
            Run cheetah run
          </Header>
          <p>
            Visit and explore CCF's <b>Cheetah Museum</b> at the
            comfort of your home.
          </p>
          <Button size="huge">Explore</Button>
        </Container>
          {/* <Segment inverted vertical as="footer">
            by @Tomiko.
          </Segment> */}
      </Segment>
    )
  }

  centerSegmentStyleClassName() {
    const centerSegmentStyleClassNames = [
      "centerSegmentStyle1",
      "centerSegmentStyle2",
      "centerSegmentStyle3",
      "centerSegmentStyle4",
    ]

    const idx = Math.floor(Math.random() * centerSegmentStyleClassNames.length);

    return centerSegmentStyleClassNames[idx];
  }
}

export default LandingPageGateway;
