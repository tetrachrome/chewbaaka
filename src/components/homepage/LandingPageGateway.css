/**
 * LandingPageGateway.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 28, 2020
 */

.ui.vertical.center.aligned.segment.LandingPageGatewayOuterContainer {
  height: 100vh;
  background-size: cover !important;
  background-position: center;
}

.ui.vertical.segment.LandingPageGatewayBackgroundStyle1 {
  background-image: url('./assets/background_CCF_01.jpg');
}

.ui.vertical.segment.LandingPageGatewayBackgroundStyle2 {
  background-image: url('./assets/background_CCF_02.jpg');
}

.ui.vertical.segment.LandingPageGatewayBackgroundStyle3 {
  background-image: url('./assets/background_CCF_03.jpg');
}

.ui.vertical.segment.LandingPageGatewayBackgroundStyle4 {
  background-image: url('./assets/background_CCF_04.jpg');
}

.ui.container.LandingPageGatewayCoreContentContainer {
  height: 300px;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: auto;
}

.ui.container.LandingPageGatewayCoreContentContainer h1.header {
  font-size: 36px;
  color: #F8F8FF;
  text-shadow: 2px 2px 4px #000000;
}

.ui.container.LandingPageGatewayCoreContentContainer p.LandingPageGatewaySubtitle {
  font-size: 21px;
  line-height: 1.6;
  font-weight: 400;
  color: #FFFFFF;
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
  text-shadow: 2px 2px 4px #000000;
}
