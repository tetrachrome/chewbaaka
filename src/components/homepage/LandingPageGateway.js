/**
 * LandingPageGateway.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Aug 04, 2020
 */

import React from 'react'

import {
  Button,
  Container,
  Header,
  Segment
} from 'semantic-ui-react'

import 'semantic-ui-css/semantic.min.css'

import './LandingPageGateway.css'

const __TEST__ = ( process.env.NODE_ENV === "test" );

export default class LandingPageGateway extends React.Component {

  static _COMPONENT_NAME_ = "landingPageGateway";

  static _TITLE_ = "landingPageGatewayTitle";

  static _SUBTITLE_ = "landingPageGatewaySubtitle";

  render() {
    const title = this.props.config.components[LandingPageGateway._COMPONENT_NAME_].constants[LandingPageGateway._TITLE_];
    const subtitle = this.props.config.components[LandingPageGateway._COMPONENT_NAME_].constants[LandingPageGateway._SUBTITLE_];
    return (
      <Segment
        className={["LandingPageGatewayOuterContainer", this.centerSegmentStyleClassName()].join(" ")}
        vertical
        textAlign="center"
        data-testid="LandingPageGatewayComponentTestId"
      >
        <Container className="LandingPageGatewayCoreContentContainer">
          <Header inverted as="h1">
            {title}
          </Header>
          <p
            className="LandingPageGatewaySubtitle"
            data-testid="LandingPageGatewayComponentSubtitlePartTestId"
          >
            {subtitle}
          </p>
          <a href="#Explore">
            <Button
              basic
              inverted
              size="huge"
            >
              Explore
            </Button>
          </a>
        </Container>
      </Segment>
    )
  }

  centerSegmentStyleClassName() {
    const centerSegmentStyleClassNames =
      this.props.config.components[LandingPageGateway._COMPONENT_NAME_].dynamicStyleGroups["centerSegmentStyleClassNames"];

    let idx = 0;

    if ( __TEST__ ) {
      // We want determinate behavior in test mode
      // because of snapshot testing.
      idx = 0;
    } else {
      idx = Math.floor(Math.random() * centerSegmentStyleClassNames.length);
    }

    return centerSegmentStyleClassNames[idx];
  }
}
