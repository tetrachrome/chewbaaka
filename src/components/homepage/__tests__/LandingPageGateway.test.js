/**
 * LandingPageGateway.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 08, 2020
 */

import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import LandingPageGateway from '../LandingPageGateway';

test('renders landing page gateway', () => {
  render(<LandingPageGateway />);

  // Tests main title element to be present.
  const titleElement = screen.getByText("Run cheetah run");
  expect(titleElement).toBeInTheDocument();

  // Tests subtitle element to be present.
  const subtitleElement = screen.getByTestId("subtitle");
  expect(subtitleElement).toBeInTheDocument();
});

test('LandingPageGateway component snapshot', () => {
  const tree = renderer
    .create(<LandingPageGateway />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
