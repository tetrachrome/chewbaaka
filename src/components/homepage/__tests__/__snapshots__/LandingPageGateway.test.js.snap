// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPageGateway component snapshot 1`] = `
<div
  className="ui vertical center aligned segment centerSegmentStyle1"
  data-testid="LandingPageGateway"
>
  <div
    className="ui container content"
  >
    <h1
      className="ui inverted header"
    >
      Run cheetah run
    </h1>
    <p
      className="subtitle"
      data-testid="subtitle"
    >
      Visit and explore CCF's 
      <b>
        Cheetah Museum
      </b>
       at the comfort of your home.
    </p>
    <a
      href="#Explore"
    >
      <button
        className="ui huge basic inverted button"
        onClick={[Function]}
      >
        Explore
      </button>
    </a>
  </div>
</div>
`;
